OpenCDMi/OCDM.cpp

```diff
@@ -171,6 +156,11 @@ namespace Plugin {
             subSystem->Set(PluginHost::ISubSystem::NOT_DECRYPTION, nullptr);
             subSystem->Release();
         }
+        RPC::IRemoteConnection* connection(_service->RemoteConnection(_connectionId));
```

Comment:
Suggest to first get the RPC::IRemoteConnection* connection(_service->RemoteConnection(_connectionId)); before you do the final Release() as the release might give you a race condition, whteher or not you can get the interface and thus, if you do not get it, stop of the container is still being called but not in the context of this thread.. So suggest:
`
    RPC::IRemoteConnection*` connection(_service->RemoteConnection(_connectionId));

    uint32_t result = _opencdmi->Release();
    ASSERT(result == Core::ERROR_DESTRUCTION_SUCCEEDED);

    PluginHost::ISubSystem* subSystem = service->SubSystems();

    subSystem->Set(PluginHost::ISubSystem::NOT_DECRYPTION, nullptr);
            subSystem->Release();
    }
    if (connection != nullptr) {
            connection->Terminate();
            connection->Release();
    }
`